### ISSUE TITLE
[LINT_BUG]: Handle non-syntactic function and object names (back ticks)

### ISSUE THREAD

**Issue by SaintRod** [Appsilon/box.linters#147]
### box.linters version

0.10.5

### Sample source code to lint

Assume test.R is a file with the below box import statement and 01_example.R is some file that exports stuff.

Sometimes it's useful to add a numeric prefix to indicate the order of operations.

The below `box::use` call doesn't fail but the box.linters does.

```
box::use(
  . / `01_example`[...],
)
```

### Lint command used

```
box::use(
  box.linters[box_default_linters],
  here[here],
  lintr[lint],
)

lint(
  filename = here("test.R"),
  linters = box_default_linters$box_unused_attached_mod_linter
)
```

### Lint result

```
Error: Linter 'box_unused_attached_mod_linter' failed in ./test.R: <text>:1:5: unexpected input
1: ./01_
        ^
```

### Expected result

If box doesn't return an error, then its linters probably shouldn't either.

If the numeric prefix naming convention isn't recommended, then that's what the lint should say.

**Comment by SaintRod** [comment]
PS. I'm not using box.linters with rhino

**Comment by caldwellst** [comment]
I came here to post the same error. This is not because of numeric prefixing, but because of the use of backticks, which are required because starting a variable name with numbers is not valid in R.

Note that this also fails using the `{box}` [recommended testing approach](https://klmr.me/box/articles/testing.html), where `__test__` modules are nested in every module and are called in the module `__init__.R` file like so:

```r
if (is.null(box::name())) {
    box::use(./`__tests__`)
}
```

**Comment by radbasa** [comment]
Will take a look at supporting back ticks in the `box::use()` calls.

**Comment by radbasa** [comment]
Non-syntactic (backticks) examples:

### Function names

#### SPECIALs

```R
box::use(
  magrittr[`%>%`]
)

c(1, 2, 3) %>% sum()
```

```R
box::use(
  path/module[`**`],
)

# where
# `**` <- function(x, y) {
#   x ^ y
# }

2 ** 4
```

#### SYMBOL_FUNCTION_CALLs

```R
box::use(
  path/module[`1_function`],
)

# where
# `1_function` <- function(x, y) {
#   x ^ y
# }

`1_function`(2, 4)
```

```R
box::use(
  path/module[...],
)

# where
# `1_function` <- function(x, y) {
#   x ^ y
# }

`1_function`(2, 4)
```

### Module names

```R
box::use(
  path/`1_module`
)

`1_module`$some_function()
```

```R
box::use(
  path/`1_module`[a],
)

a()
```

```R
box::use(
  path/`1_module`[...]
)

a()
```

### XML analysis

```R
c(1,2) %>% sum()
`01_function`(2, 3)
```

`%>%` is a SPECIAL
`\`01_function\``() is a SYMBOL_FUNCTION_CALL
```
